relations/src/r1cs/constraint_system.rs

```diff
@@ -334,19 +333,23 @@ impl<F: Field> ConstraintSystem<F> {
             }
             outlined_lc.compactify();
 
-            let mut should_promote = false;
+            let mut should_outline = false;
 
+            // Check if outlining is worthwhile.
             let this_used_times = used_times[index.0];
             let this_len = outlined_lc.len();
 
             if this_used_times * this_len > this_used_times + 2 + this_len {
-                should_promote = true;
+                should_outline = true;
             }
 
-            if should_promote {
+            // If outlining is worthwhile,
+            if should_outline {
+                // Add a new witness (the value of the linear combination).
```

Comment:
Why not use `self.new_witness` here? It would avoid errors from manual management of witness variables.